### ISSUE TITLE
[ElementWise]: elementwise inputs must have same dimensions or follow broadcast rules (input dimensions were [2,160,128] and [1,2,128]).

### ISSUE THREAD

**Issue by PhilCuriosity** [NVIDIA/TensorRT#1485]
## Description

Hello, I encountered this problem when I transferred to TRT on Jetson NX.
## Environment
**Jetpack 4.5.1**
**PyTorch Version (if applicable)**:  1.6
**Baremetal or Container (if so, version)**: 

## Relevant Files
https://drive.google.com/drive/folders/1KQ7HRaFs38XoKJ30kUxh3iSyEfqho07J?usp=sharing

## Steps To Reproduce

```
import tensorrt as trt
import sys
def ONNX_build_engine(trt_model_name='./ctpn-sim.trt',onnx_model_name='./ctpn-sim.onnx'):
  explicit_batch = 1 << (int)(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH)# trt7
  G_LOGGER = trt.Logger(trt.Logger.VERBOSE)
  # G_LOGGER = trt.Logger(trt.Logger.WARNING)
  # G_LOGGER = trt.Logger()

  batch_size = 10
  with trt.Builder(G_LOGGER) as builder, builder.create_network(explicit_batch) as network, trt.OnnxParser(network, G_LOGGER) as parser:
    builder.max_batch_size = batch_size
    builder.max_workspace_size = 1 << 30
    print('Loading ONNX file from path {}...'.format(onnx_model_name))
    with open(onnx_model_name, 'rb') as model:
        print('Beginning ONNX file parsing')
        #b = parser.parse(model.read())
        if not parser.parse(model.read()):
            for error in range(parser.num_errors):
                print(parser.get_error(error))
    if 1:
        print('Completed parsing of ONNX file')
        print('Building an engine from file {}; this may take a while...'.format(onnx_model_name))
 
        ####
        #builder.int8_mode = True
        #builder.int8_calibrator = calib
        builder.fp16_mode = False
        ####

        print("num layers:",network.num_layers)
        last_layer = network.get_layer(network.num_layers - 1) 
        network.mark_output(last_layer.get_output(0)) 
        network.get_input(0).shape = [batch_size, 1, 256, 416] # trt7
        engine = builder.build_cuda_engine(network)
        print("engine:",engine)
        with open(trt_model_name, "wb") as f:
            f.write(engine.serialize())
        print("Completed creating Engine")

**Comment by PhilCuriosity** [comment]
This is Ctpn model:
```
import torch
import torch.nn as nn
import torch.nn.functional as F
import torchvision.models as models
import ctpn_config

class basic_conv(nn.Module):
    def __init__(self,
                 in_planes,
                 out_planes,
                 kernel_size,
                 stride=1,
                 padding=0,
                 dilation=1,
                 groups=1,
                 relu=True,
                 bn=True,
                 bias=True):
        super(basic_conv, self).__init__()
        self.out_channels = out_planes
        self.conv = nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride, padding=padding, dilation=dilation, groups=groups, bias=bias)
        self.bn = nn.BatchNorm2d(out_planes, eps=1e-5, momentum=0.01, affine=True) if bn else None
        self.relu = nn.ReLU(inplace=True) if relu else None

    def forward(self, x):
        x = self.conv(x)
        if self.bn is not None:
            x = self.bn(x)
        if self.relu is not None:
            x = self.relu(x)
        return x

class CTPN_Res18_Model(nn.Module):
    def __init__(self, nc=1):
        super().__init__()
        base_model = models.resnet18(pretrained=True)
        base_model.conv1= nn.Conv2d(nc, 64, kernel_size=(7, 7), stride=(1, 1), padding=(3, 3), bias=False)
        layers = list(base_model.children())[:-2]
        self.base_layers = nn.Sequential(*layers)  # block5_conv3 output
        self.rpn = basic_conv(512, 512, 3, 1, 1, bn=False)
        self.brnn = nn.GRU(512,128, bidirectional=True, batch_first=True)
        self.lstm_fc = basic_conv(256, 512, 1, 1, relu=True, bn=False)
        self.rpn_class = basic_conv(512, 10 * 2, 1, 1, relu=False, bn=False)
        self.rpn_regress = basic_conv(512, 10 * 2, 1, 1, relu=False, bn=False)

    def forward(self, x):
        print(x.shape)  # 10, 1, 256, 416
        x = self.base_layers(x)
        # rpn
        x = self.rpn(x)    #[b, c, h, w]
        print(f'x.shape {x.shape}')
        x1 = x.permute(0,2,3,1).contiguous()  # channels last   [b, h, w, c]
        b = x1.size()  # b, h, w, c 
        x1 = x1.reshape(b[0]*b[1], b[2], b[3])

        x2, _ = self.brnn(x1)  ## This is where the bug occurred.
        # for i in self.brnn(x1):
        #    print(f'self.brnn(x1).shape {i.shape}') [16, 26, 256]  [2, 16, 128]


        xsz = x.size() 
        x3 = x2.reshape(xsz[0], xsz[2], xsz[3], 256)  # torch.Size([4, 20, 20, 256]) 
        x3 = x3.permute(0,3,1,2).contiguous()  # channels first [b, c, h, w]
        x3 = self.lstm_fc(x3)
        x = x3

        cls = self.rpn_class(x)
        regr = self.rpn_regress(x)

        cls = cls.permute(0,2,3,1).contiguous()
        regr = regr.permute(0,2,3,1).contiguous() 
        cls = cls.reshape(cls.size(0), cls.size(1)*cls.size(2)*10, 2)
        regr = regr.reshape(regr.size(0), regr.size(1)*regr.size(2)*10, 2) 

        return cls, regr
```
## This is the reason for the problem, but why?
```self.brnn = nn.GRU(512,128, bidirectional=True, batch_first=True)```

**Comment by ttyio** [comment]
Hi @PhilCuriosity , could you try JetPack 4.6, which include TensorRT 8.0.1, we have some fix for the `GRU`, thanks!

**Comment by PhilCuriosity** [comment]
Ok, thank you！
